File: Frontend/friend-nest/src/app/services/interaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { AddInteracionRequest } from '../models/request/AddInteracionRequest';
import { Observable } from 'rxjs';
import { UserInteractions } from '../models/response/UserInteractionsToPost';

@Injectable({
  providedIn: 'root',
})
export class InteractionService {
  constructor(private http: HttpClient) {}

  addInteraction(request: AddInteracionRequest): Observable<UserInteractions> {
    return this.http.post<UserInteractions>(
      'http://localhost:8080/api/addInteraction',
      request,
    );
  }
}

File: Backend/friendnest/src/main/java/pl/polsl/friendnest/service/InteractionService.java
package pl.polsl.friendnest.service;

import pl.polsl.friendnest.model.Comment;
import pl.polsl.friendnest.model.Post;
import pl.polsl.friendnest.model.User;
import pl.polsl.friendnest.model.request.AddInteractionRequest;
import pl.polsl.friendnest.model.response.UserInteractions;

public interface InteractionService {

    UserInteractions addInteraction(AddInteractionRequest interactionRequest);

    UserInteractions getUserInteractions(User user, Comment comment, Post post);
}
